src/fileoverview_comment_transformer.ts

```diff
@@ -0,0 +1,81 @@
+import * as ts from 'typescript';
+
+import * as jsdoc from './jsdoc';
+import {createNotEmittedStatement, updateSourceFileNode} from './transformer_util';
+
+/**
+ * A set of JSDoc tags that mark a comment as a fileoverview comment. These are recognized by other
+ * pieces of infrastructure (Closure Compiler, module system, ...).
+ */
+const FILEOVERVIEW_COMMENT_MARKERS: ReadonlySet<string> =
+    new Set(['fileoverview', 'externs', 'modName', 'mods', 'pintomodule']);
+
+/**
+ * A transformer that ensures the emitted JS file has an \@fileoverview comment that contains an
+ * \@suppress {checkTypes} annotation by either adding or updating an existing comment.
+ */
+export function transformFileoverviewComment(context: ts.TransformationContext) {
+  return (sf: ts.SourceFile) => {
+    let comments: ts.SynthesizedComment[] = [];
+    // Use trailing comments because that's what transformer_util.ts creates (i.e. by convention).
+    if (sf.statements.length && sf.statements[0].kind === ts.SyntaxKind.NotEmittedStatement) {
+      comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
+    }
+
+    let fileoverviewIdx = -1;
+    let parsed: {tags: jsdoc.Tag[]}|null = null;
+    for (let i = comments.length - 1; i >= 0; i--) {
+      const current = jsdoc.parseContents(comments[i].text);
+      if (current !== null && current.tags.some(t => FILEOVERVIEW_COMMENT_MARKERS.has(t.tagName))) {
+        fileoverviewIdx = i;
+        parsed = current;
+        break;
+      }
+    }
+    // Add a @suppress {checkTypes} tag to each source file's JSDoc comment,
+    // being careful to retain existing comments and their @suppress'ions.
+    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
+    // that has not been attached to some other tree node to be the file overview comment, and
+    // only applies @suppress tags from it.
+    // AJD considers *any* comment mentioning @fileoverview.
+    if (!parsed) {
+      // No existing comment to merge with, just emit a new one.
+      return addNewFileoverviewComment(sf);
+    }
+
+    // Add @suppress {checkTypes}, or add to the list in an existing @suppress tag.
+    // Closure compiler barfs if there's a duplicated @suppress tag in a file, so the tag must
+    // only appear once and be merged.
```

Comment:
In retrospect, I wonder if it woulda been better to be more strict here and reject input files that are full of nonsense.